From: Roman Gushchin <roman.gushchin@linux.dev>
Subject: mm: page_counters: put page_counter_calculate_protection() under CONFIG_MEMCG
Date: Fri, 26 Jul 2024 20:31:09 +0000

Put page_counter_calculate_protection() under CONFIG_MEMCG.

The protection functionality (min/low limits) is not supported by any
other cgroup subsystem, so page_counter_calculate_protection() and related
static effective_protection() can be compiled out if CONFIG_MEMCG is not
enabled.

Link: https://lkml.kernel.org/r/20240726203110.1577216-3-roman.gushchin@linux.dev
Signed-off-by: Roman Gushchin <roman.gushchin@linux.dev>
Acked-by: Shakeel Butt <shakeel.butt@linux.dev>
Acked-by: Johannes Weiner <hannes@cmpxchg.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Muchun Song <songmuchun@bytedance.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/page_counter.h |    6 ++++++
 mm/page_counter.c            |    2 ++
 2 files changed, 8 insertions(+)

--- a/include/linux/page_counter.h~mm-page_counters-put-page_counter_calculate_protection-under-config_memcg
+++ a/include/linux/page_counter.h
@@ -87,8 +87,14 @@ static inline void page_counter_reset_wa
 	counter->watermark = page_counter_read(counter);
 }
 
+#ifdef CONFIG_MEMCG
 void page_counter_calculate_protection(struct page_counter *root,
 				       struct page_counter *counter,
 				       bool recursive_protection);
+#else
+static inline void page_counter_calculate_protection(struct page_counter *root,
+						     struct page_counter *counter,
+						     bool recursive_protection) {}
+#endif
 
 #endif /* _LINUX_PAGE_COUNTER_H */
--- a/mm/page_counter.c~mm-page_counters-put-page_counter_calculate_protection-under-config_memcg
+++ a/mm/page_counter.c
@@ -275,6 +275,7 @@ int page_counter_memparse(const char *bu
 }
 
 
+#ifdef CONFIG_MEMCG
 /*
  * This function calculates an individual page counter's effective
  * protection which is derived from its own memory.min/low, its
@@ -446,3 +447,4 @@ void page_counter_calculate_protection(s
 			atomic_long_read(&parent->children_low_usage),
 			recursive_protection));
 }
+#endif /* CONFIG_MEMCG */
_
